### PR TITLE
Separate Prettier from ESLint

### PR DESCRIPTION
**Pull request by benjie**
## Description

Prettier is a lot faster than ESLint; integrating it with ESLint causes it to have to run in a much slower way, making saving files/autoformatting a lot clunkier.

To solve this, I recommend splitting "linting" and "formatting" into two separate tools. This PR makes these changes and integrates them with the VSCode settings.

I've also collapsed the `.eslintignore` and `.prettierignore` files into just one file - this typically makes sense to do, and is easier to maintain.

## Checklist

- [x] I have read the [**contributing**](https://github.com/ifiokjr/remirror/blob/master/docs/contributing.md) document.
- [x] My code follows the code style of this project and `yarn fix` runs successfully.
- [x] I have updated the documentation where necessary.
- [x] New code is unit tested and all current tests pass when running `yarn test` .

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -15,3 +15,16 @@ coverage
...
+**/package.json
+**/.next
+lerna.json
+.vscode
+
+/support/api/**/*.md
+/support/api/**
+/docs/api/**/*.md
+/docs/api/**
+**/temp/**
```
Comment (benjie) [inline .eslintignore]: Merged from `.prettierignore`

File: .eslintrc.js [hunk 2]
```diff
@@ -50,7 +49,6 @@ module.exports = {
     '@typescript-eslint',
     'react-hooks',
     'react',
-    'prettier',
```
Comment (benjie) [inline .eslintrc.js]: ESLint no longer runs the prettier rules, it now purely disables formatting rules that would conflict with prettier.

File: .prettierignore [hunk 3]
```diff
@@ -1,20 +0,0 @@
...
-.out
-storybook-static
-lerna.json
-examples/with-razzle/build/**
-.vscode
-
-./support/api/**/*.md
-./support/api/**
-./docs/api/**/*.md
-./docs/api/**
```
Comment (benjie) [inline .prettierignore]: No longer needed; use `.eslintignore` as the source of truth.

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,22 +1,18 @@
...
   "editor.formatOnSave": true,
   "eslint.enable": true,
   "eslint.lintTask.enable": true,
   "eslint.lintTask.options": "--ext .js,.jsx,.ts,.tsx .",
-  "prettier.disableLanguages": [
-    "javascript",
-    "javascriptreact",
-    "typescript",
-    "typescriptreact",
-  ],
```
Comment (benjie) [inline .vscode/settings.json]: VSCode now invokes prettier directly.

File: package.json [hunk 5]
```diff
@@ -63,6 +64,7 @@
     "test:e2e:ssr:watch": "yarn test:e2e:ssr --verbose=false --watch",
     "test:e2e:watch": "jest --watch --verbose=false --runInBand --config e2e/jest.config.js",
     "test:watch": "jest  --watch --verbose=false --coverage=false",
+    "prettier:run": "prettier --ignore-path .eslintignore '**/*.{js,jsx,ts,tsx,md,json,html,css,yml,yaml,graphql}'",
```
Comment (benjie) [inline package.json]: I know we don't have all these file types currently; but :man_shrugging:

**Comment by benjie** [comment]
I've added rulers to match your prettier rules; this'll help me to resize my editor splits to be the right size to avoid scrolling (currently my splits are about 100 columns, so I have to make them just a smidge wider).

**Comment by ifiokjr** [comment]
I like it. The CI build is lot faster.

**Pull request merged**